### ISSUE TITLE
Feature request: Windows Explorer integration

### ISSUE THREAD

**Issue by iahung2** [SpartanJ/ecode#180]
### Discussed in https://github.com/SpartanJ/ecode/discussions/134

Right click context menu for files and folders.

**Comment by Curculigo** [comment]
I started to feel tired of having to drag files into ecode to open them. I have to open a lot of files scattered throughout the file system. This doesn't suit the open as folder model of ecode well. Can you have a look at this feature? Thank you.

**Comment by SpartanJ** [comment]
Besides that it would be useful to associate the file extensions from ecode, why don't you set ecode as default editor for those extensions? I do that once and forget about it.

https://github.com/user-attachments/assets/d9208823-1708-4b40-a742-54e18c7f8cb1

**Comment by Curculigo** [comment]
Is ecode a single-instance application? Or will it launch a new instance for each file opened via double click in Windows Explorer?

**Comment by Curculigo** [comment]
I don't want to associate file extensions with ecode. For example, I want to view this html file on ecode but don't want it to be the default application to open html files. I can still add ecode to the Open With list, but now another problem arises. Imagine you selected 10 html files and used right click, Open With -> ecode, and 10 instances of ecode are launching. Things will quickly become a mess.

**Comment by Curculigo** [comment]
There is no integration with the file manager on both Windows and Linux/BSD. I need to view various files scattered throughout the file system. The open as folder model of ecode doesn't suit this. Currently, I have to drag these files into ecode from the file manager to open them. I'm really tired with this.

p/s: I want to ensure that just one instance of ecode is running.

**Comment by SpartanJ** [comment]
Yes, what you're looking for is issue #58 .

**Comment by ghost** [comment]
@SpartanJ Should this issue be closed?

**Comment by ghost** [comment]
@SpartanJ This is driving me mad. I don’t want to add Ecode to the file association list. Besides, the UI of Windows 11 is stupid. Even if I do so, I still have to choose `Open with` and pick Ecode from the list. Other code editors put `Open with me` directly into the context menu, so you can select a single file or a set of files and send them to the editor with right‑click → `Open with me`.

**Comment by SpartanJ** [comment]
> Other code editors put Open with me directly into the context menu, so you can select a single file or a set of files and send them to the editor with right‑click → Open with me.

Yeah, but you need a proper installer to add that. I kinda like the portable style of ecode, I'm not sure if it will be ever be an installer. Most probable outcome will be implementing #192 .
